tests/slack_notifier/test_slack_notifier.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import json
+import io
+from collections import OrderedDict
+
+from lux.utils import test
+
+__test__ = True
+
+
+class TestSlackNotifier(test.TestCase):
+    
+    
+    def setUp(self):
+        from .utils import SlackTestHandler
+        self.teststream = io.StringIO()
+        # replacing app/server loggers
+        self.logger = logging.getLogger("TestLogger")
+        self.logger.lvl = 30
+        self.handler = SlackTestHandler(self.teststream)
+        self.handler.level = 30
+        self.handler.setFormatter(logging.Formatter(json.dumps(
+                                OrderedDict([
+                                ("text", '%(name)s - %(message)s'),
+                                ("channel", "#testchannel"),
+                                ("username", "testname"),
+                                ("icon_emoji", ":testicon:")]))))
+        self.logger.addHandler(self.handler)
+    
+    def read(self, stream):
+        # the StringIO stream adds a \n at the end of every line and
```

Comment:
Tried to use BytesIO in the first place; the stream does not accept the object returned by a json.dumps() call in the formatter.